sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -110,14 +110,15 @@ abstract class QueryStageExec extends LeafExecNode {
 
   override def generateTreeString(
       depth: Int,
-      lastChildren: Seq[Boolean],
+      lastChildren: java.util.LinkedList[Boolean],
       append: String => Unit,
       verbose: Boolean,
       prefix: String = "",
       addSuffix: Boolean = false,
       maxFields: Int,
       printNodeId: Boolean,
       indent: Int = 0): Unit = {
+    lastChildren.addLast(true)
```

Comment:
Why did this `addLast` move before `super.generateTreeString`? And this function has an additional `removeLast`?

Reply:
good catch, fixed